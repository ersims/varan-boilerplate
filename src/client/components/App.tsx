import * as React from 'react';
import { hot } from 'react-hot-loader';
import Helmet from 'react-helmet';
import { Route, Switch } from 'react-router';
import favicon from '../../assets/favicon.ico';
import NavBar from './NavBar';

// Pages
import Home from '../pages/Home';
import About from '../pages/About';
import NotFound from '../pages/errors/NotFound';

class App extends React.Component {
  public render() {
    return (
      <div className="App">
        <Helmet>
          <html lang="en" />
          <title itemProp="name" lang="en">
            Varan
          </title>
          <meta charSet="utf-8" />
          <meta charSet="utf-8" />
          <meta httpEquiv="x-ua-compatible" content="ie=edge" />
          <meta name="viewport" content="width=device-width, initial-scale=1" />
          <meta name="description" content="A varan react starter app" />
          <meta name="og:type" content="website" />
          <meta name="theme-color" content="#49aa6a" />
          <link rel="icon" href={favicon} />
        </Helmet>
        <div className="AppContainer">
          <NavBar />
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/about" component={About} />
            <Route component={NotFound} />
          </Switch>
        </div>
      </div>
    );
  }
}

export default hot(module)(App);
