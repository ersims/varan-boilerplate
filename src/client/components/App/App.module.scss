@import '../../styles/mixins/fonts/open-sans';
@import '../../styles/mixins/media/media';

// Sass variables
$color-white--100: #efefef;

$color-black--100: #101010;
$color-black--80: #222222;

$color-azure-white: #dcedff;
$color-azure-white--light: lighten(desaturate($color-azure-white, 2%), 2%);
$color-azure-white--dark: darken(saturate($color-azure-white, 2%), 2%);

$color-turquoise: #59c3c3;
$color-turquoise--light: lighten(desaturate($color-turquoise, 2%), 2%);
$color-turquoise--dark: darken(saturate($color-turquoise, 2%), 2%);

$color-dark-blue: #124e78;
$color-dark-blue--light: lighten(desaturate($color-dark-blue, 2%), 2%);
$color-dark-blue--dark: darken(saturate($color-dark-blue, 2%), 2%);

.app {
  // --------------
  // App variables
  // --------------

  // NavBar
  --navbar-height: 4rem;

  // Theme
  --theme-settings-max-width: 80rem;
  --theme-settings-gutter: 2rem;

  --theme-color-primary: #{$color-turquoise};
  --theme-color-primary--contrast: #{$color-black--80};
  --theme-color-secondary: #{$color-azure-white};
  --theme-color-secondary--contrast: #{$color-black--80};
  --theme-color-tertiary: #{$color-dark-blue};
  --theme-color-tertiary--contrast: #{$color-white--100};
  --theme-color-tertiary--dark: #{$color-dark-blue--dark};

  // -------------
  // App defaults
  // -------------
  @include open-sans();
  font-size: 14px;
  @include tablet-landscape-up {
    font-size: 1.1em;
  }
  line-height: 1.5;
  ul,
  li {
    list-style-type: none;
  }
  background: var(--theme-color-primary--contrast);
}
