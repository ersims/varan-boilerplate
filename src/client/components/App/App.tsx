import React from 'react';
import { hot } from 'react-hot-loader/root';
import { Helmet } from 'react-helmet-async';
import { Route, Switch } from 'react-router';
import favicon from '../../../assets/favicon.ico';
import { Navbar } from '../Navbar/Navbar';
import { Footer } from '../Footer/Footer';

// Styles
import classes from './App.module.scss';

// Pages
import { Home } from '../../pages/Home/Home';
import { Examples } from '../../pages/Examples/Examples';
import { NotFound } from '../../pages/errors/NotFound/NotFound';

// Exports
export const App = hot(() => (
  <div className={classes.app}>
    <Helmet titleTemplate="%s | Varan">
      <html lang="en" />
      <meta charSet="utf-8" />
      <meta httpEquiv="x-ua-compatible" content="ie=edge" />
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      <meta name="description" content="A varan react starter app" />
      <meta name="og:type" content="website" />
      <meta name="theme-color" content="#59C3C3" />
      <link rel="icon" href={favicon} />
    </Helmet>
    <Navbar />
    <Switch>
      <Route exact path="/" component={Home} />
      <Route path="/examples" component={Examples} />
      <Route component={NotFound} />
    </Switch>
    <Footer />
  </div>
));
