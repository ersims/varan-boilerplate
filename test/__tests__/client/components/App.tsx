import * as React from 'react';
import App from '../../../../src/client/components/App';
import { shallow } from 'enzyme';

// Tests
test('renders correctly', () => {
  expect(shallow(<App />)).toMatchSnapshot();
});
