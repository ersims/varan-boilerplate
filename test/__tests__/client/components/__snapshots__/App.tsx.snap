// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <html
      lang="en"
    />
    <title
      itemProp="name"
      lang="en"
    >
      Varan
    </title>
    <meta
      charSet="utf-8"
    />
    <meta
      charSet="utf-8"
    />
    <meta
      content="ie=edge"
      httpEquiv="x-ua-compatible"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
    <meta
      content="A varan react starter app"
      name="description"
    />
    <meta
      content="website"
      name="og:type"
    />
    <meta
      content="#49aa6a"
      name="theme-color"
    />
    <link
      href="test-file-stub"
      rel="icon"
    />
  </HelmetWrapper>
  <div
    className="AppContainer"
  >
    <Connect(Nav) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</div>
`;
